beancount.el

```diff
@@ -448,6 +458,27 @@ With an argument move to the next non cleared transaction."
   "A list of the accounts available in this buffer.")
 (make-variable-buffer-local 'beancount-accounts)
 
+(defun beancount-inside-string-p ()
+  (nth 3 (syntax-ppss)))
+
+(defun beancount-looking-at-tag-generic (start-char start-char-string)
+  (save-excursion
+    (unless (beancount-inside-string-p)
+      ;; Loop backward until the previous character is not a valid tag
+      ;; character
+      (while (char-table-range beancount-tag-char-table (char-before))
+        (backward-char))
+      ;; Move before the start character if we hit it
+      (when (eq (char-before) start-char)
+        (backward-char))
+      (looking-at (concat start-char-string "[" beancount-tag-chars "]*")))))
+
+(defun beancount-looking-at-tag ()
+  (beancount-looking-at-tag-generic ?# "#"))
+
+(defun beancount-looking-at-link ()
+  (beancount-looking-at-tag-generic ?^ "\\^"))
```

Comment:
This functions take the same argument twice. You can use `string-to-char` or `char-to-string` instead. 

However, I would write this like
```elisp
(defun looking-at-tag ()
  (save-excursion
    (re-search-backward (concat "[^" beancount-tag-chars "]"))
    (looking-at (concat "#[" beancount-tag-chars "]"))))
```
at which point having a generic function is the same lines of code than repeating it.

Reply:
> This functions take the same argument twice. You can use string-to-char or char-to-string instead.

This is not quite true, as in `beancount-looking-at-link` the string version of the character must be escaped so that it's not treated as a regexp anchor. I initially used `char-to-string` before realizing that it did not work for links.

> However, I would write this like ... at which point having a generic function is the same lines of code than repeating it.

Agreed. Will change.

> It is probably also wise to put this kind of functions in what is conventionally the private namespace (indicated with the doubly dashed name). This has not been done consistently before, and needs to be fixed.

I had considered this but wasn't sure whether to make just the private function in my changeset with the double dash namespace or if I should match the convention in (most of) the existing code, potentially updating the full file at once in a separate PR.